OrbitCore/LinuxUtils.cpp

```diff
@@ -52,44 +53,38 @@
 namespace LinuxUtils {
 
 //-----------------------------------------------------------------------------
-std::vector<std::string> ReadProcMaps(pid_t pid) {
-  std::vector<std::string> modules;
-  // TODO: we should read the file directly instead or memory map it.
-  std::string result =
-      ExecuteCommand(absl::StrFormat("cat /proc/%d/maps", pid).c_str());
-
-  std::stringstream ss(result);
-  std::string line;
-  while (std::getline(ss, line, '\n')) {
-    modules.push_back(line);
-  }
-
-  return modules;
+outcome::result<std::vector<std::string>> ReadProcMaps(pid_t pid) {
+  OUTCOME_TRY(maps_string, OrbitCore::OrbitUtils::FileToString(
+                               absl::StrFormat("/proc/%d/maps", pid)));
+  return absl::StrSplit(maps_string, "\n");
 }
 
 //-----------------------------------------------------------------------------
-std::string ExecuteCommand(const char* a_Cmd) {
+outcome::result<std::string> ExecuteCommand(const std::string& cmd) {
   std::array<char, 128> buffer;
   std::string result;
-  std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(a_Cmd, "r"), pclose);
+  std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd.c_str(), "r"),
+                                                pclose);
   if (!pipe) {
-    LOG("Could not open pipe");
+    ERROR("Failed to execute command \"%s\"", cmd.c_str());
+    return outcome::failure(OrbitCore::Error::kFailedToExecuteCommand);
```

Comment:
I guess `outcome::failure(static_cast<std::errc>(errno))` should work as well for all syscalls (like popen).

Reply:
Yes, I think that is a great idea to use errno. 

Is that function in general an okay way to do things, or is there a better way?

Reply:
For syscalls it's the one and only error handling mechanism, so if it's a syscall, there is little you can do better.

If you call a C++-API there might be an error abstraction layer even though the error original might be triggered by a syscall.

Reply:
thanks!